[W6.1h][T12-B1]Artur Szalata

szalata opened the pull request.

szalata committed:
Add a Printable interface

OuyangDanwen commented on src/seedu/addressbook/data/person/ReadOnlyPerson.java:
```diff
@@ -82,4 +82,15 @@ default String getAsTextHidePrivate() {
+     */
+    default String getPrintableString(Printable ... printables){
+        String retString = "";
+        for (Printable p : printables){
```
Try to use a StringJoiner for string concatenation in a loop for readability and flexibility.

OuyangDanwen commented on src/seedu/addressbook/data/person/Printable.java:
```diff
@@ -0,0 +1,8 @@
+package seedu.addressbook.data.person;
+
+public interface Printable {
```
Add some comment for the interface.

OuyangDanwen commented on src/seedu/addressbook/data/person/Address.java:
```diff
@@ -55,4 +55,9 @@ public int hashCode() {
+
+    @Override
+    public String getPrintableString() {
+        return "Address: " + toString();
```
toString() can be replaced by value for readability. toString() will be called implicitly.

OuyangDanwen reviewed (changes requested):
@arturs68 Generally well done! Just one extra point: ideally the getPrintableString() method should be used somewhere.

szalata committed:
Printable: add doc to the interface
add usage of getPrintableString()
for readability replace toString() by a value so that it is called implicitly
